### ISSUE TITLE
Do not rely on automatically generated passwords in container

### ISSUE THREAD

**Issue by praiskup** [sclorg/rhscl2dockerfile#22]
We discussed this on appinfra like many months ago after first proposal of
PostgreSQL image, it is probably time to revamp that idea and make it the
best practice.

@hhorak, @mfojtik, @fweimer FYI:

I was asked whether if I could prepare the PostgreSQL so that the image
is (a) not going to require environment variable with pre-set ADMIN password
and (b) I am not going to generate the admin password automatically by
default.

Should we make this our default policy to *any* passwords for containers?
It sounds like good security safety belt.

**Comment by hhorak** [comment]
ad (a) do you mean the variable won't be required but will still exist?

**Comment by praiskup** [comment]
Yes. Just safe default.
Dne 22. 5. 2015 12:19 napsal uživatel "Honza Horak" <
notifications@github.com>:

**Comment by fweimer** [comment]
Ideally, everything would support local authentication based on UNIX domain sockets (either peer UID, or access to the path). I think it is too early to tell if this is sufficiently flexible for the container use case. It could happen that across a container boundary, all users in one container appear as a single user to the container. This may not be what is needed.

**Comment by praiskup** [comment]
@fweimer wrote:
> Ideally, everything would support local authentication based on UNIX domain
> sockets (either peer UID, or access to the path)

That is the default case implemented in PostgreSQL ATM, yes.

> I think it is too early to tell if this is sufficiently flexible for the
> container use case.

Yes, the variables to create DB + USER +PASSWORD are still there, because thats
something needed in paas and we all *want* to be consistent with.

> It could happen that across a container boundary, all users in one container
> appear as a single user to the container. This may not be what is needed.

You mean 'PostgreSQL users'?  There is a way to add additional roles.

**Comment by fweimer** [comment]
> You mean 'PostgreSQL users'? There is a way to add additional roles.

No, system users. I guess it depends on how the containers are set up, and there is no one true answer. If the kernel maps everyone in-container UID to a container-host UID, PostgreSQL will not be able to do proper authentication for separate roles using UNIX domain sockets alone.
